firefox-ios/Client/Frontend/Browser/Toolbars/Redux/ToolbarMiddleware.swift

```diff
@@ -0,0 +1,127 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Common
+import Redux
+import ToolbarKit
+
+class ToolbarMiddleware: FeatureFlaggable {
```

Comment:
you should use `final` everywhere where it is needed, since you forcefully enable static dispatching and you or other developers immediately understand that this object has no heirs and do not have to worry about the fact that your edits may break something somewhere else in code

Reply:
Thanks for your comment. We only use `final` in rare cases in our project when we want to indicate that this class should not be subclassed. In most cases adding `final` is therefor not necessary.